pre_commit/meta_hooks/check_useless_excludes.py

```diff
@@ -0,0 +1,41 @@
+import re
+import sys
+
+import pre_commit.constants as C
+from pre_commit.clientlib import load_config
+from pre_commit.git import get_all_files
+
+
+def exclude_matches_any(filenames, include, exclude):
+    include_re, exclude_re = re.compile(include), re.compile(exclude)
+    for filename in filenames:
+        if include_re.search(filename) and exclude_re.search(filename):
+            return True
+    return False
+
+
+def check_useless_excludes(config_file=None):
+    config = load_config(config_file or C.CONFIG_FILE)
```

Comment:
instead of loading the configuration, I think you can just use `ordered_load` -- then you don't have the issue of picking up the values from the manifest -- you'll of course then need to do tests for keys and can't assume they're always there -- but I think you'll be able to make stronger assertions about correctness and eliminate false positives.